src/sparseml/recipe_template/utils.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional
+
+from sparseml.pytorch.sparsification import (
+    ACDCPruningModifier,
+    GMPruningModifier,
+    MagnitudePruningModifier,
+    Modifier,
+    QuantizationModifier,
+)
+from sparseml.sparsification import ModifierYAMLBuilder
+
+
+__all__ = [
+    "ModifierBuildInfo",
+]
+
+
+class ModifierBuildInfo:
+    """
+    A class with state and helper methods for building a recipe from
+    modifier(s)
+
+    :param modifier: class of the Modifier in question
+    :param modifier_name: str A string name to be prefixed with all attribute
+        variables defined for this modifier, defaults to "mod"
+    :param fields: An Optional dictionary mapping modifier attribute names to their
+        corresponding values
+    """
+
+    def __init__(
+        self,
+        modifier: Modifier,
+        modifier_name: str = "mod",
+        fields: Optional[Dict[str, Any]] = None,
+    ):
+        self.modifier = modifier
+        self.modifier_name = modifier_name
+        self.modifier_builder = ModifierYAMLBuilder(modifier)
+        self.fields = fields or {}
+        self.__modifier_recipe_variables = {}
+        self.update(updated_fields=self.fields)
+
+    def update(self, updated_fields: Optional[Dict[str, Any]] = None):
+        """
+        Utility method to update the fields for current modifier
+
+        :param updated_fields: An Optional dictionary mapping modifier attribute names
+            to their corresponding values
+        """
+        for key, value in updated_fields.items():
+            variable_name = f"{self.modifier_name}_{key}"
+            setattr(self.modifier_builder, key, f"eval({variable_name})")
```

Comment:
are we not worried about using the (generally unsafe) `eval()` function here? Not sure what's the relationship of this class to the general UI, but if the user can manually specify the `fields` argument, using `eval()` may not be the best choice.

Reply:
I think this is putting the string eval into the recipe (which we do all over the place in recipes).

Reply:
Ahh so we could just insert the actual value here instead of evaling?

Reply:
no, what I am saying is, we would probably not like to have the user put something weird in the recipe_template config, that can later end up as an argument to `eval()` and get executed.